test/plots/linePlotTests.ts

```diff
@@ -683,4 +683,157 @@ describe("Plots", () => {
+        let data = [
+          {x: 1, y: 1},
+          {x: 2, y: 2},
+          {x: 3, y: 1},
+          {x: 4, y: 2},
+          {x: 5, y: 1}
+        ];
+        plot.addDataset(new Plottable.Dataset(data));
+
+        // Only middle point is in viewport
+        xScale.domain([2.5, 3.5]);
+
+        plot.renderTo(svg);
+        plot.croppedRenderingEnabled(true);
+
+        let path = plot.content().select("path.line").attr("d");
+        let expectedRenderedData = [1, 2, 3].map((d) => data[d]);
+        checkPathForDataPoints(path, expectedRenderedData);
+
+        svg.remove();
+      });
+
+      it("works for vertical line plots", () => {
+        let data = [
+          {x: 1, y: 1},
+          {x: 2, y: 2},
+          {x: 1, y: 3},
+          {x: 2, y: 4},
+          {x: 1, y: 5}
+        ];
+        plot.addDataset(new Plottable.Dataset(data));
+        xScale.padProportion(0);
+
+        // Only middle point is in viewport
+        yScale.domain([2.5, 3.5]);
+
+        plot.croppedRenderingEnabled(true);
+        plot.renderTo(svg);
+
+        let path = plot.content().select("path.line").attr("d");
+        let expectedRenderedData = [1, 2, 3].map((d) => data[d]);
+        checkPathForDataPoints(path, expectedRenderedData);
+
+        svg.remove();
+      });
+
+      it("adapts to scale changes", () => {
+        let data = [
+          {x: 1, y: 1},
+          {x: 2, y: 2},
+          {x: 3, y: 1},
+          {x: 4, y: 2},
+          {x: 5, y: 1}
+        ];
+        plot.addDataset(new Plottable.Dataset(data));
+
+        plot.croppedRenderingEnabled(true);
+        plot.renderTo(svg);
+
+        let path = plot.content().select("path.line").attr("d");
+        checkPathForDataPoints(path, [0, 1, 2, 3, 4].map((d) => data[d]));
+
+        // Only middle point is in viewport
+        xScale.domain([2.5, 3.5]);
+        path = plot.content().select("path.line").attr("d");
+        checkPathForDataPoints(path, [1, 2, 3].map((d) => data[d]));
+
+        // Only first point is in viewport
+        xScale.domain([-0.5, 1.5]);
+        path = plot.content().select("path.line").attr("d");
+        checkPathForDataPoints(path, [0, 1].map((d) => data[d]));
+
+        svg.remove();
+      });
+
+      function checkPathForDataPoints(path: string, data: any[]) {
+        let EPSILON = 0.0001;
+
+        let lineEdges = path.match(/(\-?\d+\.?\d*)(,|\s)(-?\d+\.?\d*)/g);
+
+        assert.strictEqual(lineEdges.length, data.length, "correct number of edges drawn");
+
+        lineEdges.forEach((edge, i) => {
+          let coordinates = edge.split(/,|\s/);
+
+          assert.strictEqual(coordinates.length, 2, "There is an x coordinate and a y coordinate");
+          assert.closeTo(xScale.invert(+coordinates[0]), data[i].x, EPSILON,
+            `Point ${i} drawn, has correct x coordinate`);
+          assert.closeTo(yScale.invert(+coordinates[1]), data[i].y, EPSILON,
+            `Point ${i} drawn, has correct y coordinate`);
```

Comment:
How do we know that data has x/y params.  Let's type data or similar?  Or we have this function be more immune to the type of data being passed in